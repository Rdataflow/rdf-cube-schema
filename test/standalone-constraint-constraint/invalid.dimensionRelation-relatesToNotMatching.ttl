@prefix dcterms: <http://purl.org/dc/terms/> .
@prefix relation: <https://cube.link/relation/> .
@prefix meta: <https://cube.link/meta/> .
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix cube: <https://cube.link/> .
@prefix observation: <https://environment.ld.admin.ch/foen/nfi/observation/max_min_undefined> .
@prefix sh: <http://www.w3.org/ns/shacl#> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix schema: <http://schema.org/> .
@base <https://example.org/> .

<cube> a cube:Cube ;
    cube:observationConstraint <shape> ;
    cube:observationSet <observationSet> .

<observationSet> cube:observation <observationA>, <observationB>, <observationC> .

<observationA> a cube:Observation ;
    cube:observedBy <observer> ;
    <dimension> 4.9 ;
    <upperConfidence> 0.1 ;
.

<shape> a cube:Constraint ;
    sh:targetClass cube:Observation ;
    sh:closed true ;
    sh:property
        [
            sh:path rdf:type ;
            sh:nodeKind sh:IRI ;
            sh:minCount 1 ;
            sh:maxCount 1
        ] ;
    sh:property
        [
            sh:path cube:observedBy ; ;
            sh:nodeKind sh:IRI ;
            sh:minCount 1 ;
            sh:maxCount 1
        ] ;
    sh:property
        [
            sh:datatype xsd:decimal ;
            sh:path <dimension> ;
            schema:name "dimension" ;
            sh:minCount 1 ;
            sh:maxCount 1 ;
        ],
        [
            sh:datatype xsd:decimal ;
            sh:path <upperConfidence> ;
            schema:name "upper confidence" ;
            sh:minCount 1 ;
            sh:maxCount 1 ;
            meta:dimensionRelation
                [
                    a relation:ConfidenceUpperBound ;
                    dcterms:type "Confidence interval" ;
                    meta:relatesTo <non-dimension> ;
                ] ;
        ];
.
