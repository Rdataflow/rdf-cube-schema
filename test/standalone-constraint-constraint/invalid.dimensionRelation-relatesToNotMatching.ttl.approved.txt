@prefix sh: <http://www.w3.org/ns/shacl#> .
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix schema: <http://schema.org/> .
@prefix cube: <https://cube.link/> .

_:report a sh:ValidationReport ;
	sh:result [
		rdf:type sh:ValidationResult ;
		sh:resultSeverity sh:Violation ;
		sh:sourceConstraintComponent sh:NodeConstraintComponent ;
		sh:sourceShape [
			sh:path sh:property ;
			sh:node <https://cube.link/shape/standalone-constraint-constraint#DimensionRelation> ;
		] ;
		sh:focusNode <https://example.org/shape> ;
		sh:resultPath sh:property ;
		sh:resultMessage "Value does not have shape <https://cube.link/shape/standalone-constraint-constraint#DimensionRelation>" ;
		sh:detail [
			rdf:type sh:ValidationResult ;
			sh:resultSeverity sh:Violation ;
			sh:sourceConstraintComponent sh:MinCountConstraintComponent ;
			sh:sourceShape [
				sh:path _:b675 ;
				sh:class cube:Constraint ;
				sh:minCount 1 ;
				sh:message "value of meta:relatesTo must be a cube dimension" ;
			] ;
			sh:focusNode _:b5 ;
			sh:resultPath _:b675 ;
			sh:resultMessage "value of meta:relatesTo must be a cube dimension" ;
		] ;
		sh:value _:b4 ;
	] ;
	sh:conforms false .

_:b675 rdf:first <https://cube.link/meta/relatesTo> ;
	rdf:rest (
		[
			sh:inversePath sh:path ;
		]
		[
			sh:inversePath sh:property ;
		]
	) .
